### ISSUE TITLE
Is there any way to use a custom icon for the battery indicator?

### ISSUE THREAD

**Issue by wayfarren** [Alexays/Waybar#3856]
Hi there, I designed a custom multi-stage battery icon for an unrelated reason, but I'd like to use it for my waybar battery indicator as well. I considered turning it into a font, which would work, but the icon has multiple colors in it, which turning the icons into a font would negate. My current theory is that I might be able to use a watermark image or album cover image module, and make it poll data from the battery percentage, and using conditionals to determine which version of the icon to display. The problem is that I have no idea how to do this. My question is this:

Is this even possible? and if so, how do I do this?

**Comment by FarrenHawk** [comment]
Update: I was able to successfully implement this. I got the icons in png format, made a really shoddy shell script to query the battery percentage and output a path to one of the images (depending on percentage, of course), then added an image in my waybar config which I put in a group with the battery percentage readout. It's not ideal, but it works.

**Comment by Alexays** [comment]
Yes, and it sounds like you already landed on the right approach 🙂 For a multi-color / multi-stage icon that a font cannot represent, the `image` module is the way: point it at a small script that queries the battery and prints the path to the matching PNG, then group it with the `battery` percentage readout.

```jsonc
"image#battery": {
    "path": "/tmp/battery-icon.png",   // or "exec" a script that prints a path
    "size": 24,
    "interval": 30
}
```

Glad you got it working — closing this out. Reopen if you want to refine it further. 🙏
